baremetal/machine_manager_test.go

```diff
@@ -0,0 +1,1563 @@
+				if savedHost.Spec.UserData == nil {
+					t.Errorf("UserData not set")
+					return
+				}
+				if savedHost.Spec.UserData.Namespace != tc.ExpectedUserDataNamespace {
+					t.Errorf("expected Userdata.Namespace %s, got %s", tc.ExpectedUserDataNamespace, savedHost.Spec.UserData.Namespace)
+				}
+				if savedHost.Spec.UserData.Name != testUserDataSecretName {
+					t.Errorf("expected Userdata.Name %s, got %s", testUserDataSecretName, savedHost.Spec.UserData.Name)
+				}
+			} else {
+				if savedHost.Spec.UserData != nil {
+					t.Errorf("did not expect user data, got %v", savedHost.Spec.UserData)
+				}
+			}
+		})
+	}
+}
+
+func TestExists(t *testing.T) {
+	scheme := runtime.NewScheme()
+	err := bmoapis.AddToScheme(scheme)
+
+	if err != nil {
+		log.Printf("AddToScheme failed: %v", err)
+	}
+
+	host := bmh.BareMetalHost{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "somehost",
+			Namespace: "myns",
+		},
+	}
+	c := fakeclient.NewFakeClientWithScheme(scheme, &host)
+
+	testCases := map[string]struct {
+		Client      client.Client
+		Machine     capi.Machine
+		FailMessage string
+		Expected    bool
+	}{
+		"Failed to find the existing host": {
+			Client: c,
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						HostAnnotation: "myns/somehost",
+					},
+				},
+			},
+			Expected:    true,
+			FailMessage: "failed to find the existing host",
+		},
+		"Found host even though annotation value incorrect": {
+			Client: c,
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						HostAnnotation: "myns/wrong",
+					},
+				},
+			},
+			Expected:    false,
+			FailMessage: "found host even though annotation value incorrect",
+		},
+		"Found host even though annotation not present": {
+			Client:      c,
+			Machine:     capi.Machine{},
+			Expected:    false,
+			FailMessage: "found host even though annotation not present",
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Logf("## TC-%s ##", name)
+		actuator, err := newMachineManager(tc.Client, nil, nil, &tc.Machine, nil)
+		if err != nil {
+			t.Error(err)
+		}
+
+		result, err := actuator.Exists(context.TODO())
+		if err != nil {
+			t.Error(err)
+		}
+		if result != tc.Expected {
+			t.Error(tc.FailMessage)
+		}
+	}
+}
+
+func TestGetHost(t *testing.T) {
+	scheme := runtime.NewScheme()
+	err := bmoapis.AddToScheme(scheme)
+
+	if err != nil {
+		log.Printf("AddToScheme failed: %v", err)
+	}
+
+	host := bmh.BareMetalHost{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "myhost",
+			Namespace: "myns",
+		},
+	}
+	c := fakeclient.NewFakeClientWithScheme(scheme, &host)
+
+	testCases := map[string]struct {
+		Client        client.Client
+		Machine       capi.Machine
+		BMMachine     capbm.BareMetalMachine
+		FailMessage   string
+		ExpectPresent bool
+	}{
+		"Did not find expected host": {
+			Client: c,
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
```

Comment:
The annotations should be on the baremetalmachine

Reply:
done